File: mavsim_matlab/parameters/control_parameters.m
% control_parameters
%
% mavsim_matlab 
%     - Beard & McLain, PUP, 2012
%     - Last updated:  
%         2/13/2019 - RWB
addpath('../chap5')
load transfer_function_coef

% AP stands for autopilot
AP.gravity = 
AP.sigma = 
AP.Va0 = 

%----------roll loop-------------
AP.roll_kp = 
AP.roll_kd = 

%----------course loop-------------
AP.course_kp = 
AP.course_ki = 

%----------sideslip loop-------------
AP.sideslip_ki = 
AP.sideslip_kp = 

%----------pitch loop-------------
AP.pitch_kp = 
AP.pitch_kd = 
K_theta_DC = 

%----------altitude loop-------------
AP.altitude_kp = 
AP.altitude_ki = 
AP.altitude_zone = 

%---------airspeed hold using throttle---------------
AP.airspeed_throttle_kp = 
AP.airspeed_throttle_ki = 

File: mavsim_python/parameters/control_parameters.py
import numpy as np
import design_projects.chap05.model_coef as TF
import parameters.aerosonde_parameters as MAV


#### TODO #####
gravity = MAV.gravity  # gravity constant
Va0 = TF.Va_trim
rho = 0 # density of air
sigma = 0  # low pass filter gain for derivative

#----------roll loop-------------
# get transfer function data for delta_a to phi
wn_roll = 0
zeta_roll = 0
roll_kp = 0
roll_kd = 0

#----------course loop-------------
wn_course = 0
zeta_course = 0
course_kp = 0
course_ki = 0

#----------yaw damper-------------
yaw_damper_p_wo = 0
yaw_damper_kr = 0

#----------pitch loop-------------
wn_pitch = 0
zeta_pitch = 0 
pitch_kp = 0
pitch_kd = 0
K_theta_DC = 0

#----------altitude loop-------------
wn_altitude = 0
zeta_altitude = 0
altitude_kp = 0
altitude_ki = 0
altitude_zone = 0

#---------airspeed hold using throttle---------------
wn_airspeed_throttle = 0
zeta_airspeed_throttle = 0
airspeed_throttle_kp = 0
airspeed_throttle_ki = 0
